### ISSUE TITLE
False-positives when comparing provenant with scancode-toolkit output

### ISSUE THREAD

**Issue by MQueiros** [mstykow/scancode-rust#1391]
First of all, let me say I highly appreciate this effort of drastically speeding-up scancode you guys are doing here.

## Summary

provenant produces a lot more false positives compared to scancode-toolkit

## Steps to reproduce

I'll use perl as an example.

```bash 
wget https://www.cpan.org/src/5.0/perl-5.38.4.tar.gz
# 1) Execute scancode-toolkit
uvx --from "scancode-toolkit==32.5.0" scancode --copyright --only-findings --json-pp output_scancode.json perl-5.38.4/ --processes 22
# 2) Execute provenant using `--compat-mode scancode`
./provenant/provenant --copyright --only-findings --json-pp output_provenant_compat_mode.json perl-5.38.4/ --processes 22 --compat-mode scancode
```

## Expected behavior

Scancode outputs `(c) 2022 Unicode(r), Inc.` and `Unicode(r), Inc.`. It might be missing some cases.

```json
{
      "path": "perl-5.38.4/lib/unicore/NamesList.txt",
      "type": "file",
      "copyrights": [
        {
          "copyright": "(c) 2022 Unicode(r), Inc.",
          "start_line": 11,
          "end_line": 11
        }
      ],
      "holders": [
        {
          "holder": "Unicode(r), Inc.",
          "start_line": 11,
          "end_line": 11
        }
      ],
      "authors": [
        {
          "author": "reversal of A862 @ Vowel A866 PHAGS-PA LETTER EE Mongolian, Chinese, Uyghur",
          "start_line": 24671,
          "end_line": 24674
        }
      ],
      "scan_errors": []
    }
```

## Actual behavior

provenant reports a lot more cases but most of them seem noisy / false-positives.

```json
{
      "path": "perl-5.38.4/lib/unicore/NamesList.txt",
      "type": "file",
      "name": "NamesList.txt",
      "base_name": "NamesList",
      "extension": ".txt",
      "size": 1671590,
      "package_data": [],
      "detected_license_expression": null,
      "detected_license_expression_spdx": null,
      "license_detections": [],
      "copyrights": [
        {
          "copyright": "(c) 2022 Unicode(r), Inc.",
          "start_line": 11,
          "end_line": 11
        },
        {
          "copyright": "copyright - 2117 x (circled latin capital letter c - 24B8) x (copyleft symbol - 1F12F) x (mask work symbol - 1F1AD) 00AA FEMININE ORDINAL INDICATOR",
          "start_line": 613,
          "end_line": 617
        },
        {
          "copyright": "(copyright sign - 00A9) x (circled latin capital letter p - 24C5) 2118 SCRIPT CAPITAL P",
          "start_line": 13484,
          "end_line": 13486
        },
        {
          "copyright": "(copyright sign - 00A9) 0043 24B9 CIRCLED LATIN CAPITAL LETTER D",
          "start_line": 15314,
          "end_line": 15316
        },
        {
          "copyright": "copyright - 2117 0050 24C6 CIRCLED LATIN CAPITAL LETTER Q",
          "start_line": 15342,
          "end_line": 15344
        },
        {
          "copyright": "(copyright sign - 00A9) x (circled latin capital letter m - 24C2) @ Regional indicator symbols @+ These characters can be used in pairs to represent regional codes. In some emoji implementations, certain pairs may be recognized and displayed by alternate",
          "start_line": 49888,
          "end_line": 49891
        },
        {
          "copyright": "(copyright sign - 00A9) x (circled latin capital letter m - 24C2)",
          "start_line": 49888,
          "end_line": 49891
        }
      ],
      "holders": [
        {
          "holder": "Unicode(r), Inc.",
          "start_line": 11,
          "end_line": 11
        },
        {
          "holder": "x (circled latin capital letter c - 24B8) x (copyleft symbol - 1F12F) x (mask work symbol - 1F1AD) 00AA FEMININE ORDINAL INDICATOR",
          "start_line": 613,
          "end_line": 617
        },
        {
          "holder": "sign - 00A9 x (circled latin capital letter p - 24C5) 2118 SCRIPT CAPITAL P",
          "start_line": 13484,
          "end_line": 13486
        },
        {
          "holder": "sign - 00A9 0043 24B9 CIRCLED LATIN CAPITAL LETTER D",
          "start_line": 15314,
          "end_line": 15316
        },
        {
          "holder": "24C6 CIRCLED LATIN CAPITAL LETTER Q",
          "start_line": 15342,
          "end_line": 15344
        },
        {
          "holder": "sign - 00A9 x (circled latin capital letter m - 24C2) @ Regional indicator symbols @+ These characters can be used in pairs to represent regional codes. In some emoji implementations, certain pairs may be recognized and displayed by alternate",
          "start_line": 49888,
          "end_line": 49891
        }
      ],
      "authors": [],
      "urls": [],
      "for_packages": [],
      "scan_errors": []
    }
```

## Environment

- provenant 1.0.8 from https://github.com/getprovenant/provenant/releases/download/v1.0.8/provenant-linux-x86_64.tar.gz

## Additional context

Summary of `./provenant/provenant compare --scancode-json output_scancode.json --provenant-json output_provenant_compat_mode.json` is [summary.json](https://github.com/user-attachments/files/31747357/summary.json).

I actually see a lot more noise when comparing copyrights of other tools, for example copyrights showing up inside HTML:

`- <div align="center"><p class="copyright">Copyright © 2000-2022 dist.authors.html AUTHORS`

when analyzing valgrind, but wanted to share a simple example first

Actually, zip 3.0 might be a good example as well:

## Copyright text of zip 3.0 using scancode

```text
Copyright (c) 1990-1997 Mark Adler, Richard B. Wales, Jean-loup Gailly, Onno van der Linden, Kai Uwe Rommel, Igor Mandrichenko, John Bush and Paul Kienitz
Copyright (c) 1990-1999 Info-ZIP.
Copyright (c) 1990-2007 Info-ZIP.
Copyright (c) 1990-2008 Info-ZIP
Copyright (c) 1995 Mark Adler
Copyright (c) 1995-1998 Jim Luther and Apple Computer, Inc.
Copyright (c) 1995-2005 Jean-loup Gailly, Brian Raiter and Gilles Vollant
Copyright (c) 1996 Scott Field
Copyright (c) 1997-2008 Info-ZIP.
Copyright (c) 1997-2008 Info-ZIP. .LP Currently
Copyright (c) 1999 Info-ZIP Chris Herborth (chrish@pobox.com)
Copyright (c) 1999-2007 Info-ZIP Chris Herborth (chrish@pobox.com) Ruslan Nickolaev (nruslan@hotbox.ru)
Copyright (c) Info-ZIP 1997 - 2008
Copyright 1992-1998 Apple Computer, Inc.
Copyrighted FREEWARE.
Portions copyright 1995 Jim Luther
copyright (c) Copyright 199x NONAME, INC.
copyright (c) Info-ZIP Group
copyright (c) Julian R Seward
copyrighted by Info-ZIP
copyrighted by the Info-ZIP group
```

## Copyright text of zip 3.0 using provenant

```text
1. Update copyrights to 2008. zip.c, zipcloak.c, zipfile.c, zipnote.c, zipsplit.c, zipup.c, README (Ed) 2. Update Who. proginfo/infozip.who (Ed) January 30th 2008 version 3.0g14 -------------------- 1. Update copyrights. fileio.c, globals.c, revision.h, util.c, zip.h, win32/win32.c, win32/win32zip.c (Ed) 2. Updates. README, proginfo/infozip.who (Ed)
8. Update copyright and ftp address in several files (Christian) 9. The RISCOS port uses '.' as directory separator, not '/' (Christian) 10. win32/makefile.bor: more options to compile the asm CRC code (Christian)
Copyright (c) 1990-1997 Mark Adler, Richard B. Wales, Jean-loup Gailly, Onno van der Linden, Kai Uwe Rommel, Igor Mandrichenko, John Bush and Paul Kienitz.
Copyright (c) 1990-2008 Info-ZIP
Copyright (c) 1990-2008 Info-ZIP. All rights reserved.
Copyright (c) 1995 Mark Adler
Copyright (c) 1995-1998 Jim Luther and Apple Computer, Inc. All rights reserved.
Copyright (c) 1995-2005 Jean-loup Gailly, Brian Raiter and Gilles Vollant.
Copyright (c) 1996 Scott Field
Copyright (c) 1997-2008 Info-ZIP.
Copyright (c) 1997-2008 Info-ZIP. .LP Currently
Copyright (c) 1999-2007 Info-ZIP Chris Herborth (chrish@pobox.com) Ruslan Nickolaev (nruslan@hotbox.ru)
Copyright (c) Info-ZIP 1997 - 2008
Copyright 1992-1998 Apple Computer, Inc.
Copyright © 1999 Info-ZIP Chris Herborth (chrish@pobox.com)
Copyrighted FREEWARE.
NOTE on copyright history: Previous versions of this source package (up to version 2.8) were
Portions copyright 1995 Jim Luther
changes to Unicode in other sections, update copyright at bottom, and some small changes to wording and examples. man/zip.1, zip.txt (Ed) 18. Put #ifdef WIN32 around WIN32 blocks. zipfile.c (Ed) May 14th 2007 version 3.0f35 ------------------------- 1. Update VMS to include new options. vms/cmdline.c, vms/zip_cli.cld (SMS)
nlm_COPYRIGHT = Op Copyright '$(copyright)' linkop = $+$(linkop)$- Caseexact linkop = $+$(linkop)$- Nod
char _copyright[] = "(c) Info-ZIP Group";
copyright (c) Julian R Seward
copyright and put @CodeSize and @DataSize into
copyright for files modified in 2002 (Onno) January 19th 2002 version 2.4e
copyrighted by Info-ZIP;
copyrighted by the Info-ZIP group
copyrights (Ed) 2. Changed text in command line messages from zip 2.4 to zip 3.0 (Ed) 3. Changes to many files for Zip64 wrapped in ifdef
extern ZCONST char *copyright[1]; extern ZCONST char * far swlicense[50]; extern ZCONST char * far versinfolines[7];
```

**Comment by mstykow** [comment]
Final verification after merging the full follow-up series through #1429:

- Verified on `main` at `ac434987f2d0a8715d82199d1470d68632ea7d84`.
- Re-ran the issue reproduction with `--copyright --only-findings --compat-mode scancode`: `NamesList.txt` now reports exactly `(c) 2022 Unicode(r), Inc.`, holder `Unicode(r), Inc.`, no authors, and no scan errors.
- Confirmed all cited Unicode-data, Zip changelog/code, and Valgrind HTML false-positive signatures are absent.
- Re-scanned Perl 5.38.4, Zip 3.0, musl, Dancer2, and Valgrind with all relevant detections; all five completed without scan errors.
- Re-ran direct ScanCode comparisons. The final Perl, Zip, and Dancer2 review queues are byte-identical to the already audited queues; remaining deltas are justified Provenant improvements or cosmetic normalization differences.
- Confirmed deterministic license ordering across four independent Zip scans.
- The final `main` CI run is green in every lane: https://github.com/getprovenant/provenant/actions/runs/33753637140

The last real-world false positive found during the final sweep (`INTEGER REFERENCES (id)` interpreted as a Dancer2 author) was fixed generically in #1429 before declaring the investigation complete.

**Comment by mstykow** [comment]
Thanks for your report @MQueiros ! I've taken this opportunity to make a detailed study of the repos you mentioned and fixed as many issues as possible. I'm about to cut another release. Please let me know afterwards if it improved for you and feel free to point us to other bugs/issues. 🙏🏼
